### PR TITLE
Anchor Links

### PR DESCRIPTION
**Pull request by Raq-HEE**
PR for Anchor Links components.

This drops a hidden div with a heading where an anchor link component has been placed. The rest of the renderring is picked up javascript (https://github.com/Manifesto-Digital/hee-prototypes/pull/27).

### CODE REVIEW

**Comment by smanivasag** [comment]
Hi @Raq-HEE - Great work 👍 

Could you take a look at the following and let me know what you think ?
- Both AnchorLink Id and Text seem to include text from tags containing `nhsuk-u-visually-hidden` class. Could we exclude them if possible ?
  The following heading snippet:
  ```
    <span class="nhsuk-u-visually-hidden">Non-urgent advice:</span>
    test title2
  ```
  creates the following AnchorLink currently:
  ```
  <a href="#non-urgenttest title23">Non-urgent advice:<br>test title2</a>
  ```
  Also, could we avoid `<br>` tag on the text if possible ?
- AnchorLinks currently seem to be created for Right Hand Content Blocks as well and not sure if this may confuse the visitor as the jumped position would show both Main and Right Hand Content Blocks.
- I think the AnchorLink for the title of the Related Content section might just be enough and AnchorLinks for each of Related Content Cards may not be required.

Thank you!

File: repository-data/application/src/main/resources/hcm-config/namespaces/hee/HomePage.yaml [hunk 1]
```diff
@@ -71,7 +71,7 @@ definitions:
           /contentBlocks:
             jcr:primaryType: frontend:plugin
             caption: Content Blocks
-            compoundList: hippostd:html,hippogallerypicker:imagelink,hee:StatementBlock,hee:YellowAlertBlock,hee:ActionLink,hee:contentCards
+            compoundList: hee:ActionLink,hee:anchorLinks,hee:contentCards,hippostd:html,hippogallerypicker:imagelink,hee:StatementBlock,hee:YellowAlertBlock
```
[the remark this answers is not in the document]
Comment (Raq-HEE) [reply repository-data/application/src/main/resources/hcm-config/namespaces/hee/HomePage.yaml]: Hey Selva - I've removed from Home page for now. Can easily add back in if required at later date.

**Comment by Raq-HEE** [comment]
>     * Both AnchorLink Id and Text seem to include text from tags containing `nhsuk-u-visually-hidden` class. Could we exclude them if possible ?

Added fix in main.js

**Comment by Raq-HEE** [comment]
>     * Also, could we avoid `<br>` tag on the text if possible ?

Added fix in main.js

**Comment by Raq-HEE** [comment]
>     * AnchorLinks currently seem to be created for Right Hand Content Blocks as well and not sure if this may confuse the visitor as the jumped position would show both Main and Right Hand Content Blocks.

Added fix in main.js. As a quick fix, I'm checking if the heading appears in right hand side by checking the class on it's ancestors. Let me know if we need to be more elegant, e.g. adding a more relevant class, or id.

**Comment by Raq-HEE** [comment]
>     * I think the AnchorLink for the title of the Related Content section might just be enough and AnchorLinks for each of Related Content Cards may not be required.

Added fix in main.js

**Comment by Raq-HEE** [comment]
>     * The following heading snippet:
>       ```
>         <span class="nhsuk-u-visually-hidden">Non-urgent advice:</span>
>         test title2
>       ```
>       
>       
>           
>             
>           
>       
>             
>           
>       
>           
>         
>       creates the following AnchorLink currently:
>       ```
>       <a href="#non-urgenttest title23">Non-urgent advice:<br>test title2</a>
>       ```

I wasn't sure about how to recreate this one. Would excluding nhsuk-u-visually-hidden also resolve this?

**Comment by smanivasag** [comment]
> > ```
> > * Both AnchorLink Id and Text seem to include text from tags containing `nhsuk-u-visually-hidden` class. Could we exclude them if possible ?
> > ```
> 
> Added fix in main.js

Thanks @Raq-HEE - Could still see tags with `nhsuk-u-visually-hidden` classes are being picked up (though they are not displaying due to the `nhsuk-u-visually-hidden` class, so may be it's Okay ?)

**Comment by smanivasag** [comment]
> > ```
> > * The following heading snippet:
> >   ```
> >     <span class="nhsuk-u-visually-hidden">Non-urgent advice:</span>
> >     test title2    
> >     
> >   creates the following AnchorLink currently:
> >   ```
> >   <a href="#non-urgenttest title23">Non-urgent advice:<br>test title2</a>
> >   ```
> > ```
> 
> I wasn't sure about how to recreate this one. Would excluding nhsuk-u-visually-hidden also resolve this?

An example of this could be found on http://localhost:8080/site/patient-and-public-information/introduction page on the local instance (`test title2` StatementBlock). Thanks!

**Comment by smanivasag** [comment]
@Raq-HEE - Observed that the anchor links seem to include tags around the titles (e.g. `<strong>CLA Licence Plus</strong>`). Is it possible to strip them off ? (e.g. `CLA Licence Plus` & `Copyright Act & ‘Library Privilege’` anchor links on  http://localhost:8080/site/patient-and-public-information/introduction on the local instance). Thanks!

**Comment by Raq-HEE** [comment]
> Could still see tags with `nhsuk-u-visually-hidden` classes are being picked up

Were these in the console log? The log was printing the array of all headings before I did the filter. Have removed the console logging.

**Comment by Raq-HEE** [comment]
> An example of this could be found on http://localhost:8080/site/patient-and-public-information/introduction page on the local instance (`test title2` StatementBlock). Thanks!

Thanks, have added a fix in main.js

**Comment by Raq-HEE** [comment]
> 
> 
> @Raq-HEE - Observed that the anchor links seem to include tags around the titles (e.g. `<strong>CLA Licence Plus</strong>`). Is it possible to strip them off ? (e.g. `CLA Licence Plus` & `Copyright Act & ‘Library Privilege’` anchor links on http://localhost:8080/site/patient-and-public-information/introduction on the local instance). Thanks!

Thanks, Fixed in main.js

**Review by smanivasag — APPROVED** [review]

**Pull request merged**